File: KASPAR GUI/src/runners/SequenceRunner.java
package runners;

import data.Action;
import data.Robot;
import data.Sequence;
import data.SequenceAction;
import gui.GuiLogger;
import java.util.Collection;
import java.util.HashSet;
import java.util.logging.Level;

public class SequenceRunner {

    private Robot robot;

    public SequenceRunner(Robot robot) {
        this.robot = robot;
    }

    public void execute(Sequence sequence) {
        if (sequence.isParallelable()) {
            ParallelRunner pr = new ParallelRunner(robot);
            Collection<Action> actions = new HashSet<Action>();
            for (SequenceAction action : sequence.getSequenceActions()) {
                actions.add(action.getAction());
            }

            pr.execute(actions);
        } else {
            ActionRunner ar = new ActionRunner(robot);
            for (SequenceAction action : sequence.getSequenceActions()) {
                ar.execute(action.getAction());

                try {
                    Thread.sleep(action.getWaitAfter());
                } catch (InterruptedException ex) {
                    GuiLogger.getLogger().log(Level.SEVERE, null, ex);
                    return;
                }
            }
        }
    }
}

File: KASPAR_python/Robot/runner/sequenceRunner.py
# import logging
# from actionRunner import ActionRunner
# 
# class SequenceRunner(object):
# 
#     def __init__(self, robot):
#         self._robot = robot
#         self._logger = logging.getLogger(__name__)
# 
#     def execute(self, sequence):
#         ar = ActionRunner(self._robot)
#         for orderedAction in sorted(sequence.actions, key=lambda a: a.order):
#             ar.execute(orderedAction.action)
# #             try {
# #                 Thread.sleep(action.getWaitAfter());
# #             } catch (InterruptedException ex) {
# #                 GuiLogger.getLogger().log(Level.SEVERE, null, ex);
# #                 return;
# #             }
